backends/tc/backend.cpp

```diff
@@ -446,14 +447,27 @@ void ConvertToBackendIR::postorder(const IR::P4Table *t) {
                 }
                 tableDefinition->addAction(actionDef, tableFlag);
             }
+            action_list.push_back(action->toString());
         }
         updateDefaultHitAction(t, tableDefinition);
         updateDefaultMissAction(t, tableDefinition);
         updateMatchType(t, tableDefinition);
         tcPipeline->addTableDefinition(tableDefinition);
+        actionsPerTable.emplace(t, action_list);
     }
 }
 
+const IR::P4Table *ConvertToBackendIR::getTableForAction(cstring action) const {
+    for (auto entry : actionsPerTable) {
```

Comment:
If one action is used in multiple tables, then it is duplicated by frontend to make it unique.
So, referring name.originalName will be incorrect. Using toString might be fine as it uses externalName() from Declaration class, please check.

Reply:
Changed the approch to get table_name